src/zenml/cli/login.py

```diff
@@ -895,20 +951,67 @@ def login(
             )
     else:
         # If no server argument is provided, and the client is not currently
-        # connected to any non-local server, we default to logging in to ZenML
-        # Pro.
-        cli_utils.declare(
-            "No server argument was provided. Logging to ZenML Pro...\n"
-            "Hint: You can run 'zenml login --local' to start a local ZenML "
-            "server and connect to it or 'zenml login <server-url>' to connect "
-            "to a specific ZenML server. If you wish to login to a ZenML Pro "
-            "server, you can run 'zenml login --pro'."
-        )
-        connect_to_pro_server(
-            api_key=api_key_value,
-            pro_api_url=pro_api_url,
-            verify_ssl=verify_ssl,
-        )
+        # connected to any non-local server, show the interactive login menu.
+        login_method = _display_login_menu()
+        
+        if login_method == "local":
+            # Start a local ZenML server and connect to it
+            start_local_server(
+                docker=docker,
+                ip_address=ip_address,
+                port=port,
+                blocking=blocking,
+                image=image,
+                ngrok_token=ngrok_token,
+                restart=restart,
+            )
+        elif login_method == "pro":
+            # Connect to ZenML Pro
+            connect_to_pro_server(
+                api_key=api_key_value,
+                pro_api_url=pro_api_url,
+                verify_ssl=verify_ssl,
+            )
+        elif login_method == "cloud":
+            # Get custom server URL from user
+            console.print()
+            server_url = Prompt.ask(
+                "[bold purple]Enter the ZenML server URL[/bold purple]",
+                default="https://"
```

Comment:
the default here is wrong.. it should somehow validate that it starts with "http" or something and not exit until such an address is provided